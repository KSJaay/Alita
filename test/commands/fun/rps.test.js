const {Client} = require("discord.js");
const rpsCommand = require("../../../commands/fun/rps");
const logger = require("../../../logger");

jest.mock("discord.js");
jest.mock("../../../logger");

describe("rps command", () => {
  const client = new Client();

  let interaction;

  beforeEach(() => {
    interaction = {
      options: {
        getSubcommand: jest.fn(),
        getChannel: jest.fn(),
        getString: jest.fn(),
      },
      guildId: "123456789",
      reply: jest.fn(),
      guild: {
        channels: {
          cache: {
            get: jest.fn().mockReturnValue({send: jest.fn()}),
          },
        },
      },
      user: {
        id: "987654321",
        username: "test",
      },
    };

    interaction.options.getString.mockReturnValue("rock");
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should reply with a success message", async () => {
    await rpsCommand.execute(client, interaction, {});

    expect(interaction.reply).toHaveBeenCalledWith(
      expect.objectContaining({
        embeds: [expect.any(Object)],
        ephemeral: false,
      })
    );
  });

  it("should call logger.error if an error is thrown", async () => {
    interaction.reply.mockImplementation(() => {
      throw new Error("test");
    });

    await rpsCommand.execute(client, interaction, {});

    expect(logger.error).toHaveBeenCalledWith(
      `Error executing 'rps' command!`,
      {
        label: "Command",
        message: "test",
        data: {},
      }
    );
  });

  it("snaphot of interaction data", () => {
    expect(rpsCommand.interaction).toMatchSnapshot();
  });
});
