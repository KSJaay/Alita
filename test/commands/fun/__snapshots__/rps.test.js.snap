// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rps command snaphot of interaction data 1`] = `
{
  "description": "Play a game of rock, paper, scissors.",
  "name": "rps",
  "options": [
    {
      "choices": [
        {
          "name": "Rock",
          "value": "rock",
        },
        {
          "name": "Paper",
          "value": "paper",
        },
        {
          "name": "Scissors",
          "value": "scissors",
        },
      ],
      "description": "Choose your weapon?",
      "name": "weapon",
      "required": true,
      "type": 3,
    },
  ],
}
`;
